File: app/src/main/java/com/example/pokedex/data/local/LocalDataSource.java
package com.example.pokedex.data.local;

import android.util.Log;
import androidx.lifecycle.LiveData;

import com.example.pokedex.data.local.entity.DetailPokemonEntity;
import com.example.pokedex.data.local.entity.PokemonEntity;
import com.example.pokedex.data.local.room.PokemonDao;

import java.util.List;

import javax.inject.Inject;

public class LocalDataSource {
  private PokemonDao pokemonDao;

  @Inject
  public LocalDataSource(PokemonDao pokemonDao) {
    this.pokemonDao = pokemonDao;
  }

  public LiveData<List<DetailPokemonEntity>> getCaughtPokemon() {
    return pokemonDao.getCaughtPokemon();
  }

  public void setCaughtPokemon(DetailPokemonEntity pokemonEntity, boolean newState,
      String nickname) {
    String oldName = pokemonEntity.getName();
    String newName = oldName;

    if (nickname.isEmpty() || nickname.equals(" ")) {
      pokemonEntity.setName(oldName);
    }
    else newName = nickname;
    pokemonEntity.setNickname(nickname);
    pokemonEntity.setName(newName);
    pokemonEntity.setCaught(newState);
    pokemonDao.update(pokemonEntity);
  }

  public LiveData<List<PokemonEntity>> getListPokemon() {
    return pokemonDao.getListPokemon();
  }

  public void insertPokemon(List<PokemonEntity> pokemonEntity) {
    pokemonDao.insertPokemon(pokemonEntity);
  }

  public void insertPokemonDetail(List<DetailPokemonEntity> pokemonEntities) {
    pokemonDao.insertPokemonDetail(pokemonEntities);
  }

  public LiveData<DetailPokemonEntity> getDetailPokemon(int id) {
    return pokemonDao.getDetailPokemon(id);
  }
}

File: core/src/main/java/com/example/data/local/LocalDataSource.kt
package com.example.data.local

import javax.inject.Inject
import com.example.data.local.room.PokemonDao
import androidx.lifecycle.LiveData
import com.example.data.local.entity.DetailPokemonEntity
import com.example.data.local.entity.PokemonEntity

class LocalDataSource @Inject constructor(private val pokemonDao: PokemonDao) {
  val caughtPokemon: LiveData<List<DetailPokemonEntity>>
    get() = pokemonDao.caughtPokemon

  fun setCaughtPokemon(
    pokemonEntity: DetailPokemonEntity, newState: Boolean,
    nickname: String
  ) {
    val name = pokemonEntity.name
    if (nickname.isEmpty() || nickname == " ") {
      pokemonEntity.name = name
    }
    pokemonEntity.nickname = nickname
    pokemonEntity.isCaught = newState
    pokemonDao.update(pokemonEntity)
  }

  val listPokemon: LiveData<List<PokemonEntity>>
    get() = pokemonDao.listPokemon

  fun insertPokemon(pokemonEntity: List<PokemonEntity>) {
    pokemonDao.insertPokemon(pokemonEntity)
  }

  fun insertPokemonDetail(pokemonEntities: List<DetailPokemonEntity>) {
    pokemonDao.insertPokemonDetail(pokemonEntities)
  }

  fun getDetailPokemon(id: Int): LiveData<DetailPokemonEntity> {
    return pokemonDao.getDetailPokemon(id)
  }
}